nemo/collections/asr/models/enhancement_models.py

```diff
@@ -0,0 +1,440 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import os
+import tempfile
+from typing import Dict, List, Optional, Union
+
+import librosa
+import soundfile as sf
+import torch
+from omegaconf import DictConfig
+from pytorch_lightning import Trainer
+from tqdm import tqdm
+
+from nemo.collections.asr.data import audio_to_audio_dataset, audio_to_text_dataset
+from nemo.collections.asr.models.audio_processing_model import AudioProcessingModel
+from nemo.collections.asr.parts.utils.audio_utils import ChannelSelectorType
+from nemo.core.classes.common import PretrainedModelInfo, typecheck
+from nemo.core.neural_types import AudioSignal, LengthsType, NeuralType
+from nemo.utils import logging
+from nemo.utils.decorators import experimental
+
+__all__ = ['EncMaskDecAudioProcessingModel']
+
+
+@experimental
+class EncMaskDecAudioProcessingModel(AudioProcessingModel):
+    """Base class for encoder-mask-decoder audio processing models."""
```

Comment:
1. This model looks like 'complete' why 'Base'?
2. May be change the file name or class name to reflect similar information.